src/groupByKey.ts

```diff
@@ -2,6 +2,18 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: string): GroupsMap<T> {
+  const map: GroupsMap<T> = {};
+
+  for (const item of items) {
+    const groupKey = item[key];
```

Comment:
This line may throw a TypeScript error if `item` does not have an index signature. You should ensure that the `key` provided is actually a key of `T`. One way to enforce this at compile time is to use a generic type that extends a type with a string index signature, or to use keyof T.